File: 090. Implementa una API Reactiva con Spring WebFlux y MongoDB con Kotlin/spring-webflux-mongo-reactive-kotlin/src/main/kotlin/com/example/service/BookService.kt
package com.example.service

import com.example.model.Book
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface BookService {
    fun findAll(): Flux<Book>
    fun findAllByAuthor(author: String): Flux<Book>
    fun findById(id: String): Mono<Book>

    fun create(book:Book): Mono<Book>
    fun update(book:Book): Mono<Book>

    fun deleteById(id: String): Mono<Void>
    fun deleteAll(): Mono<Void>


}

File: 024. Desarrolla una aplicación con Spring Boot y Angular/spring-angular/frontend/src/app/services/book.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Book} from "../models/book";

const baseUrl = 'http://localhost:8080/api/books';
@Injectable({
  providedIn: 'root'
})
export class BookService {

  constructor(private http: HttpClient) {
  }

  findAll(): Observable<Book[]> {
    return this.http.get<Book[]>(baseUrl)
  }

  findById(id: any): Observable<Book> {
    return this.http.get(`${baseUrl}/{id}`)
  }

  create(book: Book): Observable<any>{
    return this.http.post(baseUrl, book)
  }

  update(book: Book): Observable<any>{
    return this.http.put(baseUrl, book)
  }

  deleteById(id: any): Observable<any>{
    return this.http.delete(`${baseUrl}/{id}`)
  }

  deleteAll(): Observable<any>{
    return this.http.delete(`${baseUrl}`)
  }

}
